src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,60 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Deadline task for tasks that have an end date.
+ */
+public class Deadline extends Task {
+    public static final String USAGE_TEXT = "Usage: deadline <task name> /by <deadline> ";
+
+    String by;
+    LocalDate byDate;
+
+    /**
+     * Default constructor for a new task.
+     *
+     * @param name task name.
+     * @param by   deadline for deadline task.
+     */
+    public Deadline(String name, String by) {
+        this(name, false, by);
+    }
+
+    /**
+     * Full constructor with all fields exposed.
+     *
+     * @param name task name.
+     * @param done boolean state of task done.
+     * @param by   deadline for deadline task.
```

Comment:
I think there are extra whitespaces between the "by" and the "deadline", perhaps you could delete the extra whitespaces?